jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt

```diff
@@ -47,6 +48,13 @@ interface AutoDetectableToolType<VersionScheme : Version> : ToolType<VersionSche
 }
 
 interface ManagedToolType<VersionScheme : Version> : ToolType<VersionScheme> {
+    override val id: String
+        get() = telemetryId.toString()
+
+    /**
+     * Used to identify this tool with the telemetry stack
+     */
+    val telemetryId: ToolId
```

Comment:
is it even worth differentiating between the two?

Reply:
possibly not but I didn't want to leak the `ToolId` enum all the way through given most of the time all we need is a String ...